pkg/vaultutil/vaultutil.go

```diff
@@ -54,24 +56,63 @@ func InitVaultClient(addr, roleID, secretID string) (*vault.Client, error) {
 }
 
 // WriteOutputs takes any output values from a Terraform apply and then writes them into Vault
-func WriteOutputs(client *vault.Client, secretInfo VaultSecret, data map[string]tfexec.OutputMeta) error {
+func WriteOutputs(client *vault.Client, secretInfo VaultSecret, data map[string]tfexec.OutputMeta, kvVersion string) error {
 	log.Printf("Writing Output values from Terraform Apply to %s in Vault", secretInfo.Path)
-	secretData := make(map[string]interface{})
+	secretData := make(VaultKvData)
 
 	for k, v := range data {
-		secretData[k] = v.Value
+		if v.Value[0] == '"' {
+			// unquote the value as hashicorp seems to like quotes
+			value, err := strconv.Unquote(string(v.Value[:]))
+			if err != nil {
+				return err
+			}
+			secretData[k] = value
+		} else {
+			secretData[k] = string(v.Value[:])
+		}
```

Comment:
this unquote looks strange, where is the additional quote coming from? should also have a test case for this.

Reply:
Good question. I looked into this a bit more and it appears to be an oddity with how `terraform output` works, see this issue: https://github.com/hashicorp/terraform/issues/26831. According to the issue, using `tf output -json` should result in no quotes being included, I will see if I can do the same using the `tfexec` wrapper.

Reply:
From what I can see, terraform calls this "working as intended". I have updated the `TestWriteVaultOutputs` test to add an unquoted string for validation

Reply:
In qontrace reconcile, we use `tf output -json` [code](https://github.com/app-sre/qontract-reconcile/blob/7701ef69c3bbba05047862ac4b250413aaa09040/reconcile/utils/lean_terraform_client.py#L93), should we keep the behaviour same here?

Reply:
trace to source [here](https://github.com/hashicorp/terraform-exec/blob/1b14c363512a1d4edd6ce6850375f24e461b910a/tfexec/output.go#L39-L66), it's already json output, I don't understand why there is an additional quote, please inspect where it's coming from.

Reply:
I'm thinking now that it may be some bug in tfexec. I was playing around with the [localstack test tf repo I made](https://gitlab.cee.redhat.com/rywallac/tf-repo-tekton-localstack-test) using the terraform CLI directly on my machine and got this output:

```
terraform output -json
{
  "vpc_id": {
    "sensitive": false,
    "type": "string",
    "value": "vpc-4d808e4a"
  }
}
```

No extra quote there unlike what I saw with tfexec. Investigating this further.

Reply:
I'm going to dry a different approach with handling the `json.RawMessage` values here, stand by

Reply:
@hemslo the extra quotes are coming from here: https://github.com/hashicorp/terraform-exec/blob/main/tfexec/cmd.go#L205. Since `v` being passed uses `json.RawMessage` types for `v.Type` & `v.Value`, when the output values are decoded, the quotes are included since `json.RawMessage` is just a [raw byte array](https://pkg.go.dev/encoding/json#RawMessage) of JSON values, not a string

Reply:
I added a comment trying to describe this behavior and also not completely fail if something cannot be unquoted since outputs could theoretically be any valid JSON object

Reply:
cc @jonmosco

Reply:
Then should we always deserialize it first, like this

```golang
var value interface{}
json.Unmarshal(v.Value, &value)
```

Reply:
Thanks, that was exactly what I should've done originally. Appreciate the help! Looks like this method also doesn't add any extra quotes.